src/main/java/de/tum/in/www1/artemis/service/exam/ExamUserService.java

```diff
@@ -0,0 +1,134 @@
+package de.tum.in.www1.artemis.service.exam;
+
+import java.awt.Rectangle;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.text.PDFTextStripperByArea;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import com.nimbusds.oauth2.sdk.util.StringUtils;
+
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.exam.ExamUser;
+import de.tum.in.www1.artemis.repository.ExamUserRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.FileService;
+import de.tum.in.www1.artemis.web.rest.dto.ExamUsersNotFoundDTO;
+import de.tum.in.www1.artemis.web.rest.dto.ImageDTO;
+import de.tum.in.www1.artemis.web.rest.errors.InternalServerErrorException;
+
+/**
+ * Service Implementation for managing Exam Users.
+ */
+@Service
+public class ExamUserService {
+
+    private final Logger log = LoggerFactory.getLogger(ExamUserService.class);
+
+    private final ExamUserRepository examUserRepository;
+
+    private final UserRepository userRepository;
+
+    private final FileService fileService;
+
+    public ExamUserService(FileService fileService, UserRepository userRepository, ExamUserRepository examUserRepository) {
+        this.examUserRepository = examUserRepository;
+        this.userRepository = userRepository;
+        this.fileService = fileService;
+    }
+
+    /**
+     * Extracts all images from PDF file and map each image with student registration number.
+     *
+     * @param file PDF file to be parsed
+     * @return list of ExamUserWithImageDTO
+     */
+    public List<ExamUserWithImageDTO> parsePDF(MultipartFile file) {
+
+        try (PDDocument document = PDDocument.load(file.getBytes())) {
+            ImageExtractor imageExtractor = new ImageExtractor(document);
+            imageExtractor.process();
+            List<ImageDTO> images = imageExtractor.getImages();
+            List<ExamUserWithImageDTO> studentWithImages = new ArrayList<>();
+
+            for (ImageDTO image : images) {
+                PDFTextStripperByArea stripper = new PDFTextStripperByArea();
+                stripper.setSortByPosition(true);
+
+                // A4 page, where the Y coordinate of the bottom is 0 and the Y coordinate of the top is 842.
```

Comment:
Why 842?

Reply:
A4 page measures 842 user units,  and i use it to calculate the image coordinates.